subsys/net/lib/nrf70_fw_ext/CMakeLists.txt

```diff
@@ -48,3 +48,38 @@ if(DEFINED CONFIG_NRF_WIFI_PATCHES_EXT_FLASH_DISABLED OR
     ${gen_dir}/nrf70.bin.inc
   )
 endif()
+
+# If PM is enabled then sysbuild will handle the patches
+if(NOT CONFIG_PARTITION_MANAGER AND CONFIG_NRF_WIFI_PATCHES_EXT_FLASH_STORE)
+  dt_nodelabel(qspi_nodelabel IMAGE ${DEFAULT_IMAGE} NODELABEL "qspi")
+  dt_reg_addr(qspi_xip_address IMAGE ${DEFAULT_IMAGE} PATH "${qspi_nodelabel}" NAME "qspi_mm")
+
+  dt_nodelabel(nrf70_fw_node NODELABEL "nrf70_fw_partition")
+  if(NOT DEFINED nrf70_fw_node)
+    message(FATAL_ERROR "nrf70_fw_partition node not found in dts")
+  endif()
+  dt_prop(nrf70_fw_offset PATH ${nrf70_fw_node} PROPERTY reg INDEX 0)
+  if(NOT DEFINED nrf70_fw_offset)
+    message(FATAL_ERROR "nrf70_fw_partition.reg not found in dts")
+  endif()
+  math(EXPR NRF70_PATCH_ADDR "${qspi_xip_address} + ${nrf70_fw_offset}")
+
+  # Using python module instead of intelhex command line tool to handle windows
+  execute_process(
+    COMMAND ${Python3_EXECUTABLE}
+      -c "import sys; import intelhex; intelhex.bin2hex(sys.argv[1], sys.argv[2], int(sys.argv[3]))"
+      ${NRF70_PATCH}
+      ${CMAKE_BINARY_DIR}/zephyr/nrf70.hex
+      ${NRF70_PATCH_ADDR}
+    RESULT_VARIABLE _bin2hex_result
+    OUTPUT_VARIABLE _bin2hex_output
+    ERROR_VARIABLE _bin2hex_error
+    COMMAND_ECHO STDOUT
+  )
+
+  if(NOT _bin2hex_result EQUAL 0)
+    message(FATAL_ERROR "Failed to convert binary patch to hex format: \n${_bin2hex_output}\n${_bin2hex_error}")
+  endif()
+  set_property(GLOBAL APPEND PROPERTY HEX_FILES_TO_MERGE ${CMAKE_BINARY_DIR}/zephyr/nrf70.hex ${CMAKE_BINARY_DIR}/zephyr/zephyr.hex)
+  set_property(TARGET runners_yaml_props_target PROPERTY hex_file ${CMAKE_BINARY_DIR}/zephyr/merged.hex)
```

Comment:
this just does not work, enable mcuboot and watch it fall over

Reply:
The support for external flash in `samples/wifi` never used PM, even when it was working in v2.7.0. So, it has it's limitations. As discussed, we can drop non-PM external flash support for Wi-Fi samples and extend PM subsystem to handle Wi-Fi samples.